cluster/helpers.go

```diff
@@ -54,6 +58,69 @@ func verifySig(addr string, digest []byte, sig []byte) (bool, error) {
 	return actual == expect, nil
 }
 
+// signOperator return the operator with config hash and enr EIP712 signatures by the provided k1 private key.
+func signOperator(secret *ecdsa.PrivateKey, operator Operator, configHash [32]byte) (Operator, error) {
+	var err error
+	operator.ConfigSignature, err = signEIP712(secret, operator.Address, configHash[:])
+	if err != nil {
+		return Operator{}, err
+	}
+
+	operator.ENRSignature, err = signEIP712(secret, operator.Address, []byte(operator.ENR))
+	if err != nil {
+		return Operator{}, err
+	}
+
+	return operator, nil
+}
+
+// aggSign returns a bls aggregate signatures of the message signed by all the shares.
+func aggSign(secrets [][]*bls_sig.SecretKeyShare, message []byte) ([]byte, error) {
+	var sigs []*bls_sig.Signature
+	for _, shares := range secrets {
+		for _, share := range shares {
+			secret, err := tblsconv.ShareToSecret(share)
+			if err != nil {
+				return nil, err
+			}
+			sig, err := tbls.Sign(secret, message)
+			if err != nil {
+				return nil, err
+			}
+			sigs = append(sigs, sig)
+		}
+	}
+
+	aggSig, err := tbls.Scheme().AggregateSignatures(sigs...)
+	if err != nil {
+		return nil, errors.Wrap(err, "aggregate signatures")
+	}
+
+	b, err := aggSig.MarshalBinary()
+	if err != nil {
+		return nil, errors.Wrap(err, "marshal signature")
+	}
+
+	return b, nil
+}
+
+// signEIP712 signs the config hash digest and returns the signature.
+func signEIP712(secret *ecdsa.PrivateKey, address string, message []byte) ([]byte, error) {
+	const nonce = 0
+
+	digest, err := digestEIP712(address, message, nonce)
```

Comment:
discussion: is `nonce` a required field?

Reply:
at the moment yes